test-integration/calendar.coffee

```diff
@@ -97,39 +94,37 @@ describe 'Calendar and Stats', ->
 
 
     _.each ['PHYSICS', 'BIOLOGY'], (courseCategory) =>
-      CourseSelect.goTo(@, courseCategory)
+      new CourseSelect(@).goTo(courseCategory)
 
-      @waitClick(linkText: 'Student Scores').then => @addTimeout(60)
-      @waitAnd(css: '.scores-report .course-scores-title')
-      @sleep(500)
+      wait(@).click(linkText: 'Student Scores').then => @addTimeout(60)
+      wait(@).for(css: '.scores-report .course-scores-title')
 
       # Click the "Review" links (each task-plan)
-      @forEach '.review-plan', (item, index, total) =>
+      forEach @, css: '.review-plan', (item, index, total) =>
         console.log 'opening Review', courseCategory, index, 'of', total
         item.click()
-        @waitClick(css: '.task-breadcrumbs > a')
-        @waitAnd(css: '.course-scores-wrap')
+        wait(@).click(css: '.task-breadcrumbs > a')
+        wait(@).for(css: '.course-scores-wrap')
 
       # Click each Student Forecast
-      @forEach css: '.student-name', ignoreLengthChange: true, (item, index, total) =>
+      forEach @, css: '.student-name', ignoreLengthChange: true, (item, index, total) =>
         console.log 'opening Student Forecast', courseCategory, index, 'of', total
         item.click()
-        @waitAnd(css: '.chapter-panel.weaker, .no-data-message')
-        @waitClick(css: '.performance-forecast a.back')
-        @waitAnd(css: '.course-scores-wrap')
+        wait(@).for(css: '.chapter-panel.weaker, .no-data-message')
+        wait(@).click(css: '.performance-forecast a.back')
+        wait(@).for(css: '.course-scores-wrap')
 
       # only test the 1st row of each Student Response
-      @forEach '.fixedDataTableRowLayout_rowWrapper:nth-of-type(1) .task-result', (item, index, total) =>
+      forEach @, css: '.fixedDataTableRowLayout_rowWrapper:nth-of-type(1) .task-result', (item, index, total) =>
```

Comment:
It could be, but i'm having trouble envisioning what that would look like.  Maybe an `enum` interface that could be used like: `enum(@).each(css: '.foo')` ?